packages/office-ui-fabric-react/src/components/DetailsList/DetailsColumn.tsx

```diff
@@ -235,16 +238,25 @@ export class DetailsColumn extends BaseComponent<IDetailsColumnProps> {
   }
 
   private _onDragStart(item?: any, itemIndex?: number, selectedItems?: any[], event?: MouseEvent): void {
-    if (itemIndex && this.props.setDraggedItemIndex) {
-      this.props.setDraggedItemIndex(itemIndex);
+    if (itemIndex) {
+      this._updateHeaderDragInfo(itemIndex);
       this._root.current.classList.add(styles.borderWhileDragging);
     }
   }
 
   private _onDragEnd(item?: any, event?: MouseEvent): void {
+    if (event) {
+      this._updateHeaderDragInfo(-1, event);
+    }
+    this._root.current.classList.remove(styles.borderWhileDragging);
+  }
+
+  private _updateHeaderDragInfo(itemIndex: number, event?: MouseEvent) {
     if (this.props.setDraggedItemIndex) {
-      this.props.setDraggedItemIndex(-1);
-      this._root.current.classList.remove(styles.borderWhileDragging);
+      this.props.setDraggedItemIndex(itemIndex);
+    }
```

Comment:
Should we keep this logic within `_onDragEnd` to avoid breaking backwards compat now that it is gated by `if (event)` or is the event guaranteed to be present?